feat: creating-a-regular-order-from-an-existing-cart (#879)
vtexdocs/dev-portal-content: docs/guides/Checkout/creating-a-regular-order-from-an-existing-cart.mdx

---
title: "Creating a regular order from an existing cart"
slug: "creating-a-regular-order-from-an-existing-cart"
hidden: false
createdAt: "2023-07-18T17:08:52.219Z"
updatedAt: "2023-07-18T17:08:52.219Z"
excerpt: "Learn how to use VTEX APIs to place a regular order from an existing cart."
---

This step-by-step guide will walk you through the process of using the [Checkout API](https://developers.vtex.com/docs/api-reference/checkout-api) to create an order from an existing shopping cart. This process is divided into the following steps:

1. **Get cart information:** The process begins with retrieving the `orderFormId` of the current shopping cart via the [Get current or create a new cart](https://developers.vtex.com/docs/api-reference/checkout-api#get-/api/checkout/pub/orderForm) endpoint.
2. **Place the order:** The process continues by placing the order on the VTEX platform via the [Place order from an existing cart](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/transaction) endpoint.

> 3. **Handling the order payment and processing:** The process ends by sending the necessary payment data to resolve the order payment and processing order details.

Reviewer: Sugestão:
Para manter uma padronização de referenciar links (como foi feito nos dois primeiros passos), aqui no passo 3 seria interessante adicionar os links para os endpoints de pagamento e processamento da ordem.
Ex.  [sending the necessary payment data](https://developers.vtex.com/docs/api-reference/payments-gateway-api#post-/api/pub/transactions/-transactionId-/payments)  e  [processing order](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/gatewayCallback/-orderGroup-)

> 3. **Handling the order payment and processing:** The process ends by sending the necessary payment data to resolve the order payment and processing order details.

Author: ```suggestion
3. **Handling the order payment and processing:** The process ends by [sending the necessary payment data](https://developers.vtex.com/docs/api-reference/payments-gateway-api#post-/api/pub/transactions/-transactionId-/payments) to resolve the order payment and [processing order details](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/gatewayCallback/-orderGroup-).
```

## Before you begin

Before proceeding any further, ensure that you have a valid `appKey` and `appToken` granted with the necessary permissions to access the [Checkout](https://developers.vtex.com/docs/api-reference/checkout-api) and [Payments Gateway](https://developers.vtex.com/docs/api-reference/payments-gateway-api) APIs. For more information, please refer to the [Roles](https://help.vtex.com/en/tutorial/roles--7HKK5Uau2H6wxE1rH5oRbc) article.

> > ✅ Click the steps below to navigate through this article and explore specific topics. Relevant code will be highlighted and focused for your convenience.

Reviewer: Aqui eu sugiro o mesmo _disclaimer_ do outro PR.
"Click the steps below to navigate through this article and explore specific topics. The relevant codes for each phase or parameter will be highlighted on the right side of the screen".

> > ✅ Click the steps below to navigate through this article and explore specific topics. Relevant code will be highlighted and focused for your convenience.

Author: ```suggestion
> ✅ Click the steps below to navigate through this article and explore specific topics. The relevant codes for each phase or parameter will be highlighted on the right side of the screen.
```

  "invoiceData": null,
  "customData": null,
  "itemMetadata": null,
  "hooksData": null,
  "ratesAndBenefitsData": null,
  "subscriptionData": null,
  "itemsOrdination": null
}
```

</CH.Code>

---

## Step 2 - Placing the order

Place a new order via the [Place order from an existing cart](https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForm/-orderFormId-/transaction) endpoint, providing the following information in the request body:

- `referenceId`: `orderFormId` obtained in the previous step, serving as the unique identifier (ID) of the `orderForm` associated with the shopping cart from which the order will be created.
- `savePersonalData`: set as `true` if the shopper’s data provided during checkout should be saved for future reference (default is `false`).
- `optinNewsLetter`: set as `true` if the shopper opted to receive the newsletter (default is `false`).
- `value`: total value of the order without separating cents (e.g., $24.99 is represented by `2499`).
- `referenceValue`: reference value of the order, without separating cents (e.g., $24.99 is represented by `2499`). This is used for calculating interest, if applicable.

> - `interestValue`: interest rate to be applied if applicable.

Reviewer: Adicionar vírgula antes "if applicable"

> - `interestValue`: interest rate to be applied if applicable.

Author: ```suggestion
- `interestValue`: interest rate to be applied, if applicable.
```

Upon receiving a `201 Created` response, the request will provide a response body containing the [orderForm](https://developers.vtex.com/docs/guides/orderform-fields) with comprehensive details of the newly created regular order. Take note of these pieces of information and proceed to [Resolving the order payment](#step-3---resolving-the-order-payment).

> - `orderId` - ID of the order within VTEX’s Order Management System (OMS). You can find the `orderId` within each object in the orders array.

Reviewer: citar _ordens_ como código (`orders`).

> - `orderId` - ID of the order within VTEX’s Order Management System (OMS). You can find the `orderId` within each object in the orders array.

Author: ```suggestion
- `orderId` - ID of the order within VTEX’s Order Management System (OMS). You can find the `orderId` within each object in the `orders` array.
```
